src/p4est_io.h

```diff
@@ -68,6 +108,398 @@ p4est_t            *p4est_inflate (sc_MPI_Comm mpicomm,
+ *                                  file after a successful call of this
+ *                                  function.
+ * \param [in,out] blocks           After a successful function call this
+ *                                  variable holds an array with a length
+ *                                  corresponding to the number of arrays in the
+ *                                  file that are successfully found and seeked.
+ *                                  The values in the array are the
+ *                                  number of bytes of stored data per quadrant.
+ *                                  Require elem_size->elem_size
+ *                                  == sizeof (p4est_file_block_metadata_t)
+ *                                  on input and preserve it on output.
+ *                                  See p4est_file_block_metadata_t to obtain
+ *                                  detailed information about the data blocks
+ *                                  of the file.
+ * \param [out] errcode             An errcode that can be interpreted by \ref
+ *                                  p4est_file_error_string and
+ *                                  \ref p4est_file_error_class.
+ * \return                          0 for a successful call and -1 in case of
+ *                                  an error. See also \ref errcode argument.
+ */
+int                 p4est_file_info (p4est_t * p4est, const char *filename,
+                                     char *user_string,
+                                     sc_array_t * blocks, int *errcode);
+
+/** Converts a p4est file error code into a p4est_file error class.
+ * This function turns MPI error codes into MPI error classes if
+ * MPI IO is enabled.
+ * If MPI IO is not enabled, the function processes the errors outside
+ * of MPI but passes version 1.1 errors to MPI_Error_class.
+ * Furthermore, p4est_file functions can create \ref P4EST_FILE_COUNT_ERROR
+ * as errcode what is also processed by this function.
+ * \param [in]  errcode     An errcode from a p4est_file function.
+ * \param [out] errclass    Non-NULL pointer. Filled with matching
+ *                          errclass on success.
+ * \return                  sc_MPI_SUCCESS on successful conversion.
+ *                          Other MPI error code otherwise.
+ */
+int                 p4est_file_error_class (int errcode, int *errclass);
+
+/** Turn p4est_file errcode into a string.
+ * errclass must be a class that is output by \ref p4est_file_error_class.
+ * \param [in] errclass     An errclass that is output by \ref
+ *                          p4est_file_error_class.
+ * \param [in,out] string   At least sc_MPI_MAX_ERROR_STRING bytes.
+ * \param [out] resultlen   Length of string on return.
+ * \return                   sc_MPI_SUCCESS on success or
+ *                           something else on invalid arguments.
+ */
+int                 p4est_file_error_string (int errclass, char *string,
```

Comment:
Would it be legale to pass an error code as well as an error class? If yes, please update the docs accordingly. Can we follow the convention used in MPI?

Reply:
Now, there are only error codes in p4est and they can all be handled by `p4est_file_error_string`.